File: backend/src/Domain/Entities/Occurrence.cs
﻿using Domain.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Entities
{
    public class Occurrence
    {
        public int Id { get; set; }
        public int IdPerson { get; set; }
        public int IdAddress { get; set; }
        public DateTime CreatedOn { get; set; }
        public DateTime? ExpectedDate { get; set; }
        public StatusEnum Status { get; set; }
        public int Frequency { get; set; }
        public string? Image { get; set; }
        public CategoryEnum Category { get; set; }
        public string? Desciption { get; set; }
        public virtual Person? Person { get; set; }
        public virtual Address? Address { get; set; }
        public virtual Resolution? Resolution { get; set; }
    }
}

File: frontend/src/app/models/occurrence.ts

export enum Status{
  Aware = 0,
  InProgress = 1,
  Done = 2
}

export class Address{
  id?: number;
  street?: string;
  number?: string;
  district?: string;
  city?: string;
  state?: string;

  constructor(obj: Partial<Occurrence>){
    Object.assign(this, obj);
  }
}


export class Occurrence{
  constructor(public address: Address){}
  id?: number;
  idPerson?: number;
  idAddress?: number;
  public status: number = 0;
  frequency?: number;
  image?: string;
  createdOn?: Date;
  desciption?:string;
  category?: number;
  expectedDate?: Date;
}

